### ISSUE TITLE
Error while upgrading model from 1.x to 2.0 

### ISSUE THREAD

**Issue by mr-munjal** [open-mmlab/mmdetection#3726]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

**Reproduction**
1. What command or script did you run?
```
python tools/upgrade_model_version.py --num-classes 3  old_epoch.pth  new_epoch.pth
```
2. Did you make any modifications on the code or config? Did you understand what you have modified?

Config 'faster_rcnn_x101_64x4d_fpn_1x_coco'
According to the readme in legacy_1.x folder of config. 
The num-classes provided in above command is number of classes from v1.model.
I have changed the number of classes in config file where num_classes in config of v2.model (2) is num_classes in v1.model (3) - 1.

3. What dataset did you use?

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
sys.platform: linux
Python: 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GPU 0: Tesla P4
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.6.0+cu101
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0+cu101
OpenCV: 4.1.2
MMCV: 1.1.2
MMDetection: 2.4.0+64e2d44
MMDetection Compiler: GCC 7.5
MMDetection CUDA Compiler: 10.1

2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.
```
The model and loaded state dict do not match exactly

size mismatch for rpn_head.rpn_reg.weight: copying a param with shape torch.Size([8, 256, 1, 1]) from checkpoint, the shape in current model is torch.Size([12, 256, 1, 1]).
size mismatch for rpn_head.rpn_reg.bias: copying a param with shape torch.Size([8]) from checkpoint, the shape in current model is torch.Size([12]).
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by v-qjqs** [comment]
Hi @mr-munjal , I guess the size mismatch error might be caused by your manually changed 'num-classes'. Since you modify the 'num-classes' to 3 which yields size 12 in 'rpn_head.rpn_reg.bias', generally the number of classes in rpn_head is less equal 2.

**Comment by mr-munjal** [comment]
> Hi @mr-munjal , I guess the size mismatch error might be caused by your manually changed 'num-classes'. Since you modify the 'num-classes' to 3 which yields size 12 in 'rpn_head.rpn_reg.bias', generally the number of classes in rpn_head is less equal 2.

Hey @v-qjqs , Since in old version of mmdetection we used to add 1 extra class for background so total were 3 But in this version they have removed the background part and we ought to provide the number of classes 1 less than the previous version. This is according to the Readme in configs->legacy_1.x directory.

Moreover, I have tried with the possible permutations of num-classes both in argument of the command and the config file. But, each time i receive this error.

**Comment by jonbakerfish** [comment]
I also encountered similar errors. My case is 2-class old model -> 1-class new mode:

```
size mismatch for rpn_head.rpn_reg.weight: copying a param with shape torch.Size([6, 256, 1, 1]) from checkpoint, the shape in current model is torch.Size([12, 256, 1, 1]).
size mismatch for rpn_head.rpn_reg.bias: copying a param with shape torch.Size([6]) from checkpoint, the shape in current model is torch.Size([12]).
```

**Comment by ZwwWayne** [comment]
Fixed in #3754.
